EIPS/eip-7543.md

```diff
@@ -0,0 +1,158 @@
+0xd5 DECLN   ln(a) -> b : (ac, aq, precision, steps) -> (bc, bq)
+0xd6 DECSIN sin(a) -> b : (ac, aq, precision, steps) -> (bc, bq)
+
+precision is the # of digits kept during all calculations. steps for DECEXP and DECSIN are the # of Taylor expansion steps. steps for DECLN is the depth of the continued fractions expansion.
+
+### Why these functions?
+
+The proposed functions (+,-,*,/,exp,ln,sin) form a small set that combined enable all calculation of all elementary functions, which includes the sets of sums, products, roots and compositions of finitely many polynomial, rational, trigonometric, hyperbolic, and exponential functions, including their inverse functions.
+
+a^b = exp(b * ln(a)) gives us powers and polynomials.
+cos(a) = sin(tau/4-a), tan(a)=sin(a)/cos(a), etc., gives us all of trigonometry.
+
+together with arithmetic, we get all elementary functions.
+
+### DECNEG instead of DECSUB
+
+Negation is a more general operation vs subtraction. OPCODEs should be as fundamental as possible and as complex as desirable.
+For the same reason, we have DECINV instead of DECDIV.
+
+DECSUB(a,b) = DECADD(a,DECNEG(b))
+DECDIV(a,b) = DECMUL(a,DECINV(b))
+
+### DECEXP, DECSIN via Taylor series
+
+The Taylor series of exp and sin converge everywhere and fast. The error falls as fast as the factorial of steps.
+
+### DECLN via continued fractions
+
+Ln converges fast using continued fractions within the interval ]0,2]. The implementation scales the input into this interval and scales the result back correctly.
+
+### math/big
+
+The implementation allows arbitrary precision, in theory. In practice, resources are always finite.
+
+The implementation uses the same lib as used for the stack (uint256).
+Using math/big would allow for arbitrary[*] intermediate precision. That version is also functional, on another branch.
+Even tho using math/big, input and output values have to fit into the stack.
+Using uint256 is much faster, natural for the stack and still allows for far more precision that most applications need.
+
+
+
+
+<!--
+  The Specification section should describe the syntax and semantics of any new feature. The specification should be detailed enough to allow competing, interoperable implementations for any of the current Ethereum platforms (besu, erigon, ethereumjs, go-ethereum, nethermind, or others).
+
+  It is recommended to follow RFC 2119 and RFC 8170. Do not remove the key word definitions if RFC 2119 and RFC 8170 are followed.
+
+  TODO: Remove this comment before submitting
+-->
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+## Rationale
+
+### gas
+
+all the above OPCODEs are deterministic, hence the gas cost can be determined. at the same time, the calculations are complex and depend on the input.
```

Comment:
capital letter for the sentence beginnings

Reply:
done